subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -1484,6 +1484,47 @@ static int cmd_wifi_ps_wakeup_mode(const struct shell *sh, size_t argc, char *ar
 	return 0;
 }
 
+static int cmd_wifi_set_rts_threshold(const struct shell *sh, size_t argc, char *argv[])
+{
+	struct net_if *iface = net_if_get_first_wifi();
+	int rts_threshold = -1; /* Default value if user supplies "off" argument */
+	int err = 0;
+
+	context.sh = sh;
+
+	if (strcmp(argv[1], "off") != 0) {
+		long rts_val = shell_strtol(argv[1], 10, &err);
+
+		if (err) {
+			shell_error(sh, "Unable to parse input (err %d)", err);
+			return err;
+		}
+
+		if (rts_val < WIFI_RTS_THRESHOLD_MIN) {
+			PR_WARNING("Invalid value\n");
+			shell_help(sh);
+			return -EINVAL;
+		} else if (rts_val <= WIFI_RTS_THRESHOLD_MAX) {
```

Comment:
so, we silently ignore if the value if > MAX? Any specific reason we shouldn't be throwing EINVAL ?

Reply:
Any positive value greater than the MAX threshold limit in turn will make RTS off.

Reply:
So, again the driver should have this special encoding that `-1` means it's off, as this is 802.11 protocol param, why don't we use the protocol values here?

And looking at 802.11-2020 + 802.11ax-2021: `dot11RTSThreshold` range is `0-6500631` with default value of `6500631` meaning it's disabled.